src/renderer/components/dialog-add-version.tsx

```diff
@@ -168,25 +191,32 @@ export class AddVersionDialog extends React.Component<
   }
 
   private renderPath(): JSX.Element | null {
-    const { folderPath, isValidElectron } = this.state;
+    const { isValidElectron, folderPath, existingLocalVersion } = this.state;
+    const canSwitch = isValidElectron && existingLocalVersion;
 
     if (!folderPath) return null;
-
-    const info = isValidElectron
-      ? `We found an ${getElectronNameForPlatform()} in this folder.`
-      : `We did not find a ${getElectronNameForPlatform()} in this folder...`;
-
     return (
       <Callout>
-        {info}
-        {this.renderVersionInput()}
+        {this.buildDialogText()}
+        {!canSwitch && this.renderVersionInput()}
       </Callout>
     );
   }
 
+  private buildDialogText(): string {
+    const { existingLocalVersion, isValidElectron } = this.state;
+
+    if (isValidElectron && existingLocalVersion) {
+      return `This folder is already in use as version "${existingLocalVersion.version}". Would you like to switch to that version now?`;
+    } else if (isValidElectron) {
+      return `We found an ${getElectronNameForPlatform()} in this folder.`;
+    } else {
+      return `We did not find a ${getElectronNameForPlatform()} in this folder...`;
+    }
+  }
```

Comment:
```suggestion
  private buildDialogText(): string { 
    const { isValidElectron, isValidVersion, existingLocalVersion } = this.state;
    const canAdd = isValidElectron && isValidVersion && !existingLocalVersion;
    const canSwitch = isValidElectron && existingLocalVersion;

    if (canSwitch)
      return `This folder is already in use as version "${existingLocalVersion!.version}". Would you like to switch to that version now?`;

    if (canAdd)
      return `We found an ${getElectronNameForPlatform()} in this folder.`;

    return `We did not find a ${getElectronNameForPlatform()} in this folder...`;
  } 

```

Reply:
The second block on line 214, i think the conditional must stay as Felix's old implementation 'isValidElectron'. If we change it as 'canAdd' and add 'isValidVersion' and '!existingLocalVersion' , the logic changes and we don't get the text we expect in certain test cases

I will however change the if-else styling to remove the curly brackets (is this a style choice or is it objectively better?)

Reply:
Yes, you're right about the logic. Good catch.

No, the if-else braces are subjective and you can do whatever you like. Now that you mention it, the style I used in my suggested change is probably subjectively worse since it's not as consistent with the rest of the code around it, so probably better to keep the braces.

It's arguably better to not use 'else' after 'return' since it's an unnecessary code instruction that the compiler will have to parse and then discard. How much difference that overhead makes in the real world is probably negligible, but that's the closest I know of to an _objective_ answer of what would be the better way to write that function.

tldr; probably keep the braces in, and keep the elses out.

Anyway, thanks for your patience with all the variable & function naming. :slightly_smiling_face: